### ISSUE TITLE
Error on `doom/reload`: `void-variable arg`

### ISSUE THREAD

**Issue by xiaobing-huang** [hlissner/doom-emacs#5344]
### What did you expect to happen?

M-x doom/reload, to update private config.

### What actually happened?

```
Debugger entered--Lisp error: (void-variable arg)
  (if arg " -c" "")
  (format "%s -ic '%S env%s'" (string-trim (shell-command-to-string (format "getent passwd %S | cut -d: -f7" (user-login-name)))) doom-bin (if arg " -c" ""))
  (compile (format "%s -ic '%S env%s'" (string-trim (shell-command-to-string (format "getent passwd %S | cut -d: -f7" (user-login-name)))) doom-bin (if arg " -c" "")) t)
  (set-buffer (compile (format "%s -ic '%S env%s'" (string-trim (shell-command-to-string (format "getent passwd %S | cut -d: -f7" (user-login-name)))) doom-bin (if arg " -c" "")) t))
  (save-current-buffer (set-buffer (compile (format "%s -ic '%S env%s'" (string-trim (shell-command-to-string (format "getent passwd %S | cut -d: -f7" (user-login-name)))) doom-bin (if arg " -c" "")) t)) (let ((w (get-buffer-window (current-buffer)))) (select-window w) (add-hook 'compilation-finish-functions #'(lambda (_buf status) (if (equal status "finished\n") (progn (delete-window w) (message "Successfully regenerated envvar file")) (error "Failed to generate env file"))) nil 'local)))
  (let (process-environment) (save-current-buffer (set-buffer (compile (format "%s -ic '%S env%s'" (string-trim (shell-command-to-string (format "getent passwd %S | cut -d: -f7" ...))) doom-bin (if arg " -c" "")) t)) (let ((w (get-buffer-window (current-buffer)))) (select-window w) (add-hook 'compilation-finish-functions #'(lambda (_buf status) (if (equal status "finished\n") (progn ... ...) (error "Failed to generate env file"))) nil 'local))))
  (if IS-WINDOWS (message "Can't regenerate envvar file from within Emacs in ...") (let (process-environment) (save-current-buffer (set-buffer (compile (format "%s -ic '%S env%s'" (string-trim (shell-command-to-string ...)) doom-bin (if arg " -c" "")) t)) (let ((w (get-buffer-window (current-buffer)))) (select-window w) (add-hook 'compilation-finish-functions #'(lambda (_buf status) (if ... ... ...)) nil 'local)))))
  (progn (if IS-WINDOWS (message "Can't regenerate envvar file from within Emacs in ...") (let (process-environment) (save-current-buffer (set-buffer (compile (format "%s -ic '%S env%s'" (string-trim ...) doom-bin (if arg " -c" "")) t)) (let ((w (get-buffer-window ...))) (select-window w) (add-hook 'compilation-finish-functions #'(lambda ... ...) nil 'local))))))
  (if (file-readable-p doom-env-file) (progn (if IS-WINDOWS (message "Can't regenerate envvar file from within Emacs in ...") (let (process-environment) (save-current-buffer (set-buffer (compile (format "%s -ic '%S env%s'" ... doom-bin ...) t)) (let ((w ...)) (select-window w) (add-hook 'compilation-finish-functions #'... nil 'local)))))))
  (let ((doom-reloading-p t)) (doom-run-hooks 'doom-before-reload-hook) (if (file-readable-p doom-env-file) (progn (if IS-WINDOWS (message "Can't regenerate envvar file from within Emacs in ...") (let (process-environment) (save-current-buffer (set-buffer (compile ... t)) (let (...) (select-window w) (add-hook ... ... nil ...))))))) (doom-initialize 'force) (condition-case err (progn (general-auto-unbind-keys) (unwind-protect (doom-initialize-modules 'force) (general-auto-unbind-keys t))) ((debug error) (message "PRIVATE CONFIG ERROR: %s" err) nil)) (doom-run-hooks 'doom-after-reload-hook) (message "Config successfully reloaded!"))
  (progn (delete-window w) (let ((doom-reloading-p t)) (doom-run-hooks 'doom-before-reload-hook) (if (file-readable-p doom-env-file) (progn (if IS-WINDOWS (message "Can't regenerate envvar file from within Emacs in ...") (let (process-environment) (save-current-buffer (set-buffer ...) (let ... ... ...)))))) (doom-initialize 'force) (condition-case err (progn (general-auto-unbind-keys) (unwind-protect (doom-initialize-modules 'force) (general-auto-unbind-keys t))) ((debug error) (message "PRIVATE CONFIG ERROR: %s" err) nil)) (doom-run-hooks 'doom-after-reload-hook) (message "Config successfully reloaded!")))
  (if (equal status "finished\n") (progn (delete-window w) (let ((doom-reloading-p t)) (doom-run-hooks 'doom-before-reload-hook) (if (file-readable-p doom-env-file) (progn (if IS-WINDOWS (message "Can't regenerate envvar file from within Emacs in ...") (let (process-environment) (save-current-buffer ... ...))))) (doom-initialize 'force) (condition-case err (progn (general-auto-unbind-keys) (unwind-protect (doom-initialize-modules 'force) (general-auto-unbind-keys t))) ((debug error) (message "PRIVATE CONFIG ERROR: %s" err) nil)) (doom-run-hooks 'doom-after-reload-hook) (message "Config successfully reloaded!"))) (user-error "Failed to reload your config"))
  (closure ((w . #<window 81>) t) (_buf status) (if (equal status "finished\n") (progn (delete-window w) (let ((doom-reloading-p t)) (doom-run-hooks 'doom-before-reload-hook) (if (file-readable-p doom-env-file) (progn (if IS-WINDOWS ... ...))) (doom-initialize 'force) (condition-case err (progn (general-auto-unbind-keys) (unwind-protect ... ...)) ((debug error) (message "PRIVATE CONFIG ERROR: %s" err) nil)) (doom-run-hooks 'doom-after-reload-hook) (message "Config successfully reloaded!"))) (user-error "Failed to reload your config")))(#<buffer *compilation*<.doom.d>> "finished\n")
  run-hook-with-args((closure ((w . #<window 81>) t) (_buf status) (if (equal status "finished\n") (progn (delete-window w) (let ((doom-reloading-p t)) (doom-run-hooks 'doom-before-reload-hook) (if (file-readable-p doom-env-file) (progn (if IS-WINDOWS ... ...))) (doom-initialize 'force) (condition-case err (progn (general-auto-unbind-keys) (unwind-protect ... ...)) ((debug error) (message "PRIVATE CONFIG ERROR: %s" err) nil)) (doom-run-hooks 'doom-after-reload-hook) (message "Config successfully reloaded!"))) (user-error "Failed to reload your config"))) #<buffer *compilation*<.doom.d>> "finished\n")
  compilation-handle-exit(exit 0 "finished\n")
  compilation-sentinel(#<process comint> "finished\n")
```

### Steps to reproduce

1. open Emacs
2. M-x doom/reload

### System Information

https://pastebin.com/SFkwpcxZ

**Comment by hlissner** [comment]
This was a regression caused by https://github.com/hlissner/doom-emacs/commit/c2c6a6411801b8b9ded87503b9454c512ae7f989 which should be resolved as of 24866a1. Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!

**Comment by rodelrod** [comment]
@hlissner I had this problem and after upgrading to this commit I have different error message:

```
-*- mode: compilation; default-directory: "~/" -*-
Comint started at Fri Aug  6 02:14:36

/bin/bash -ic '"/home/rodelrod/emacs-profiles/doom-emacs/bin/doom" env'
To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

mkdir: cannot create directory '/.virtualenvs': Permission denied
mkdir: cannot create directory '/.pyenv': Permission denied
mkdir: cannot create directory '/.pyenv': Permission denied

Failed to initialize virtualenvwrapper.

Perhaps pyenv-virtualenvwrapper has not been loaded into your shell properly.
Please restart current shell and try again.

bash: /.cargo/env: No such file or directory

Command 'direnv' not found, but can be installed with:

sudo snap install direnv  # version v2.18.2, or
sudo apt  install direnv  # version 2.21.2-1

See 'snap info direnv' for additional versions.

Can't find emacs in your PATH

Comint exited abnormally with code 1 at Fri Aug  6 02:14:38
```

**Comment by rodelrod** [comment]
My system information for the above error: https://pastebin.com/EVjgig2N

**Comment by xiaobing-huang** [comment]
@hlissner After upgrading to the latest commit, got below error:
```
Comint started at Fri Aug  6 09:40:25

zsh:1: command not found: getent -ic '"/Users/i074455/.emacs.d/bin/doom" env'

Comint exited abnormally with code 127 at Fri Aug  6 09:40:27

**Comment by hlissner** [comment]
Sorry for the late response: this was "resolved" in 9687c04 by simply not trying to regenerate your envvar file from `doom/reload`. It's impossible to predict or tame the environment `M-x doom/reload` may be executed in. `doom env` is for dumping your shell environment, so it should be used from the shell exclusively.

What `doom/reload` _will_ do though, is re_load_ your old envvar file, which was fixed in #5399. In any case, I'll consider this issue resolved.
